### PR TITLE
docs: add samples for using timeouts

### PR DESCRIPTION
**Pull request by olavloite**
Adds samples for running queries and updates with timeouts. This is one of the most frequently asked questions for the Spring Data Spanner project.

### CODE REVIEW

File: spanner/spring-data/src/main/java/com/example/spanner/SpannerRepositorySample.java [hunk 1]
```diff
@@ -41,5 +45,22 @@ public void runRepositoryExample() {
     int deletedLastNameCount = this.singerRepository.deleteByLastName("a last name");
   }
 
+  public void runRepositoryTimeoutExample() {
```
Comment (averikitsch) [inline spanner/spring-data/src/main/java/com/example/spanner/SpannerRepositorySample.java:48]: based on the style guide https://googlecloudplatform.github.io/samples-style-guide/, we typically have samples in individual files with separate region tags. Can you update?

**Review by averikitsch — CHANGES_REQUESTED** [review]

**Comment by averikitsch** [comment]
We also expect tests for each sample

**Comment by olavloite** [comment]
> We also expect tests for each sample

There are no tests or test infrastructure at all set up for this project, which makes adding tests for these samples difficult.

**Comment by averikitsch** [comment]
> > We also expect tests for each sample
> 
> There are no tests or test infrastructure at all set up for this project, which makes adding tests for these samples difficult.

It is a part of our policy for samples to have tests: https://googlecloudplatform.github.io/samples-style-guide/#testing. Without tests samples are not maintainable. The [api-spanner-java](https://github.com/orgs/GoogleCloudPlatform/teams/api-spanner-java) team owns the spanner samples, therefore without explicit consent from this owner we may not be able to accept these samples.

**Comment by olavloite** [comment]
> api-spanner-java

Specifically the Spring Data Spanner project is maintained by a different team.

@meltsufin Could you chime in here, please? Do you know whether there's a specific reason that there are no tests set up for these samples? Or are there any plans for doing so?

**Comment by meltsufin** [comment]
> > api-spanner-java
> 
> Specifically the Spring Data Spanner project is maintained by a different team.
> 
> @meltsufin Could you chime in here, please? Do you know whether there's a specific reason that there are no tests set up for these samples? Or are there any plans for doing so?

We have [tests](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-repository-sample/src/test/java/com/example) for Spring Data Spanner samples the we maintain in spring-cloud-gcp. We do not maintain java-docs-samples repo, however.

**Comment by olavloite** [comment]
@anguillanneuf Would you mind merging this PR? I don't think I have permission to merge anything in this repo.

File: spanner/spring-data/src/main/java/com/example/spanner/SpannerRepositoryTimeoutSample.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+// [START spanner_spring_data_repository_timeout_sample]
```
Comment (minherz) [inline spanner/spring-data/src/main/java/com/example/spanner/SpannerRepositoryTimeoutSample.java:19]: Please, do not create new region tag without registering it with http://devrel

**Review by minherz — COMMENTED** [review]
I am not familiar with the demand. There is no issue external or internal referenced here.
I would suggest to avoid adding region tags before you register it and to follow region tag guidelines. I cannot add internal go-link here. Please, search for "region tag" for internal doc.

**Comment by minherz** [comment]
@olavloite, thank you very much for your contribution. Our requirement for tests means to reduce the maintenance burden over code samples. We will be happy to help you with setting up necessary infrastructure configurations.

Before you being, please review other [spanner samples](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/main/spanner) to see what they use.
You might want to consider implementing the test as a unit test by mocking underlying infrastructure because the goal of the code sample is to be complaint with the framework's and libraries it is using and not to validate the end-to-end behavior of the sample vs. Google APIs.
Feel free to reach out by mail or Chat. I've invited you to the "Java samples review requests" space.

**Comment by olavloite** [comment]
Closing for now due to the testing discussion.

**Pull request closed without merging**